packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -314,25 +371,34 @@ export class PartialSequenceLengths {
         combinedPartialLengths: PartialSequenceLengths,
         segment: ISegment,
         removalInfo?: IRemovalInfo) {
+        const isLocal = (removalInfo === undefined && segment.seq === UnassignedSequenceNumber)
+            || (removalInfo !== undefined && segment.removedSeq === UnassignedSequenceNumber);
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        let seq = segment.seq!;
+        let seq = isLocal ? segment.localSeq! : segment.seq!;
         let segmentLen = segment.cachedLength;
         let clientId = segment.clientId;
         let removeClientOverlap: number[] | undefined;
 
         if (removalInfo) {
-            seq = removalInfo.removedSeq;
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            seq = isLocal ? segment.localRemovedSeq! : removalInfo.removedSeq;
             segmentLen = -segmentLen;
-            // this code still assume removed client id and
-            // overlap clients are separate. so we need to pull
-            // then apart first.
-            clientId = removalInfo.removedClientIds[0];
-            removeClientOverlap = removalInfo.removedClientIds.length > 1
-                ? removalInfo.removedClientIds.slice(1)
-                : undefined;
+            // The client who performed the remove is always stored
+            // in the first position of removalInfo.
+            const removingClient = removalInfo.removedClientIds[0];
+            assert(typeof removingClient === "number",
+                "Local client cannot have removed first if there are overlapping removes");
+            clientId = removingClient;
+            const hasOverlap = removalInfo.removedClientIds.length > 1;
+            removeClientOverlap = hasOverlap ? removalInfo.removedClientIds : undefined;
         }
 
-        const seqPartials = combinedPartialLengths.partialLengths;
+        const seqPartials = isLocal ?
+            combinedPartialLengths.unsequencedRecords?.partialLengths : combinedPartialLengths.partialLengths;
```

Comment:
ahh. so this seems to be part of how it work,  as we are looking at the unsequenced partial length, and not the regular. still trying to understand how we are avoiding mixing local and remote seqs